src/main/java/de/bixilon/minosoft/gui/rendering/camera/shaking/CameraShaking.kt

```diff
@@ -20,44 +20,55 @@ import de.bixilon.kutil.avg.FloatAverage
 import de.bixilon.kutil.time.TimeUtil.millis
 import de.bixilon.minosoft.config.profile.profiles.rendering.camera.shaking.ShakingC
 import de.bixilon.minosoft.gui.rendering.camera.Camera
-import de.bixilon.minosoft.gui.rendering.renderer.drawable.Drawable
 import de.bixilon.minosoft.protocol.protocol.ProtocolDefinition
+import kotlin.math.PI
 import kotlin.math.sin
 
 class CameraShaking(
     private val camera: Camera,
     private val profile: ShakingC,
-) : Drawable {
+) {
     private var rotation = 0.0f
-    private var strength = FloatAverage(5 * ProtocolDefinition.TICK_TIME * 1_000_000L, 1.0f)
-    private val speed = FloatAverage(5 * ProtocolDefinition.TICK_TIME * 1_000_000L, 0.0f)
+    private var previous_velocity = 0.0f
+    private var strength = 0.02f
+    private val speed = FloatAverage(10 * ProtocolDefinition.TICK_TIME * 1_000_000L, 0.0f)
+    private val physics = camera.context.connection.camera.entity.physics
 
     val isEmpty: Boolean get() = rotation == 0.0f
 
-    override fun draw() {
-        this.strength += 1.0f
-        val strength = this.strength.avg * profile.amplifier // strength affects how far it goes
+    fun update(): Boolean {
+        if(!this.physics.onGround){
+            this.speed += 0.0f
+            this.rotation = 0.0f
+            return false
+        }
+        val time = millis()
+        this.rotation = bobbing(time,10f,this.strength)
+        this.strength = 0.02f
+        return true
+    }
 
-        val physics = camera.context.connection.camera.entity.physics
-        val velocity = physics.velocity.xz.length2().toFloat() // velocity affects how quick it goes
-        if (velocity > 0.003 && physics.onGround) {
+    private fun bobbing(time: Long,frequency: Float, intensity: Float): Float {
+        var velocity = this.physics.velocity.xz.length2().toFloat() // velocity affects how quick it goes
+        if(this.previous_velocity != 0.0f){
+            velocity = this.previous_velocity + (velocity - this.previous_velocity) * 0.25f // interpolate
```

Comment:
there is a `interpolateLinar` function from kutil

Reply:
And I don't get that part: Why are you interpolating the previous velocity? 

This function is called every frame, whereas the velocity updates every tick. So you might interpolate the same values.

Also the purpose of the `FloatAverage` (`this.speed`) does pretty much the same, it averages a float over (here) 10 ticks. Just call `this.speed.avg` to get it.